### PR TITLE
[docs] Add breaking changes for 7.7

### PR DESCRIPTION
**Pull request by dedemorton**
Adds documentation for all PRs returned by https://github.com/elastic/beats/pulls?q=is%3Apr+is%3Aclosed+label%3A%22breaking+change%22+label%3Av7.7.0

### CODE REVIEW

**Comment by mikemadden42** [comment]
jenkins run the tests please

File: libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc [hunk 1]
```diff
@@ -12,9 +12,25 @@
...
+[float]
+====  Environment variables can no longer reference other environment variables
 
-//Description
+Environment variables are still supported in the {beats} configuration.
+However, starting in Version 7.7.0, you can no longer reference an environment
+variable that references another environment variable.
+
+For example, if you set `export NAME=${COMPANY_NAME}`, you cannot use
+`name: ${NAME}` in the configuration.
```
Comment (urso) [inline libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]: this is confusing. If in the shell I use `export NAME=${COMPANY_NAME}`, then `COMPANY_NAME` is resolved immediately and NAME has a constant string. Escaping the `$` will indeed display the problem.

Besides that, environment variables can not contain objects anymore. E.g. `export HOSTS="[a, b, c]"` used to be parsed into an array of 3. This is not the case anymore I think.

File: libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc [hunk 2]
```diff
@@ -12,9 +12,25 @@
...
+[float]
+==== Docker and kubernetes processors no longer allowed in script processor
+
+Prior to this release, it was possible to use the `add_docker_metadata` and
+`add_kubernetes_metadata` processors in the `script` processor. This was not a
+good practice because it sometimes resulted in memory and file descriptor leaks.
+
+Starting in version 7.7.0, scripts that use these processors will fail. To
+resolve this problem, define the processors in your configuration instead of the
+script.
```
Comment (urso) [inline libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:34]: @exekias Please have a look. Anything else missing?

Comment (exekias) [reply libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:34]: This looks good to me

File: libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc (hunk 1 above)
Comment (dedemorton) [reply libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]: Can I get help with an example + text that is better then?

Comment (blakerouse) [reply libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]: @urso and @dedemorton You are both correct, neither will result in previous behavior.

The simplest example is an environment variable cannot reference another environment variable or event a reference in the configuration.

So I would go with:

For example, if you have `export FOO=${BAR}` and use `var: ${FOO}` in the configuration, then the configuration will result in `var: "${BAR}"` where `${BAR}` will not be expanded.

Comment (dedemorton) [reply libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]: Sounds good! I'll have to change "foo" and "bar", though, or former editors will come back from the grave to haunt me. :-)

File: libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc [hunk 3]
```diff
@@ -16,11 +16,12 @@
...
 Environment variables are still supported in the {beats} configuration.
-However, starting in Version 7.7.0, you can no longer reference an environment
-variable that references another environment variable.
+However, starting in Version 7.7.0, an environment variable cannot reference
+another environment variable or event in the configuration.
 
-For example, if you set `export NAME=${COMPANY_NAME}`, you cannot use
-`name: ${NAME}` in the configuration.
+For example, if you have `export VAR1=${VAR2}` and use `var: ${VAR1}` in
+the configuration, the configuration results in `var: "${VAR2}"`, where
```
Comment (dedemorton) [inline libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]:
```suggestion
the configuration, this results in `var: "${VAR2}"`, where
```

Comment (dedemorton) [inline libbeat/docs/release-notes/breaking/breaking-7.7.asciidoc:23]:
```suggestion
the configuration, this results in `var: "${VAR2}"`, where
```

**Pull request merged**